OpenOversight/tests/routes/test_auth.py

```diff
@@ -509,3 +510,29 @@ def test_user_can_change_dept_pref(mockdata, client, session):
 
         user = User.query.filter_by(email=GENERAL_USER_EMAIL).one()
         assert user.dept_pref == AC_DEPT
+
+
+def test_user_email_update_resets_session_token(mockdata, client, session, faker):
+    with current_app.test_request_context():
+        _, user = login_user(client)
+        original_uuid = user.uuid
+
+        token = user.generate_email_change_token(faker.ascii_email())
+        client.get(url_for("auth.change_email", token=token), follow_redirects=True)
+
+        assert original_uuid != current_user.uuid
```

Comment:
I'm running into issues getting the Flask-Login to reload `current_user` in unit tests (but it works fine when I stand up the server locally). I think it might be an issue with the test_request_context but I'll take a closer look tomorrow.